internal/pkg/state/view.go

```diff
@@ -121,6 +122,18 @@ func (v *View) MinerSectorSize(ctx context.Context, maddr addr.Address) (abi.Sec
 	return minerState.Info.SectorSize, nil
 }
 
+// MinerSectors returns all the miner's onchain sectors
+func (v *View) MinerSectorsForEach(ctx context.Context, maddr addr.Address, f func(miner.SectorOnChainInfo) error) error {
```

Comment:
This allows us to count a miner's sectors.

Reply:
This unfortunately couples the caller to the `SectorOnChainInfo` type, preventing versioning of it. If we only need the count right now, I suggest you have the method compute that directly. Otherwise replace the single struct parameter with multiple parameters, each being a field of the struct, but only as many of them as a caller today actually needs.

Reply:
Converted to MinerSectorCount